bot/exts/utils/snekbox.py

```diff
@@ -116,16 +117,75 @@ async def convert(cls, ctx: Context, code: str) -> list[str]:
         return codeblocks
 
 
+class PythonVersionSwitcherButton(ui.Button):
+    """A button that allows users to re-run their eval command in a different Python version."""
+
+    def __init__(
+        self,
+        job_name: str,
+        version_to_switch_to: Literal["3.10", "3.11"],
+        snekbox_cog: "Snekbox",
+        ctx: Context,
+        code: str
+    ) -> None:
+        self.version_to_switch_to = version_to_switch_to
+        super().__init__(label=f"Run in {self.version_to_switch_to}", style=enums.ButtonStyle.primary)
+
+        self.snekbox_cog = snekbox_cog
+        self.ctx = ctx
+        self.job_name = job_name
+        self.code = code
+
+    async def callback(self, interaction: Interaction) -> None:
+        """
+        Tell snekbox to re-run the user's code in the alternative Python version.
+
+        Use a task calling snekbox, as run_job is blocking while it waits for edit/reaction on the message.
+        """
+        scheduling.create_task(self.snekbox_cog.run_job(self.job_name, self.ctx, self.version_to_switch_to, self.code))
+        await interaction.message.delete()
```

Comment:
Why is it important to not block? Does the Discord UI somehow tie into when this callback finishes? If it's only because the delete shouldn't wait, then that could be avoided by simply switching the order of these lines.

Reply:
Yea, blocking here causing the discord UI to say the interaction failed. Discord only acks after the callback finishes.

Reply:
Added comment in this: https://github.com/python-discord/bot/pull/2218/commits/3d9566a84ba877290efe384068bf00e3aa342d7f